src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,43 @@
 package core.basesyntax;
 
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder result = new StringBuilder();
+        result.append("Report for period ").append(dateFrom).append(" - ").append(dateTo);
+
+        for (String name : names) {
+            int salary = 0;
+            for (String datum : data) {
+                String[] parts = datum.split(" ");
+                int hours = Integer.parseInt(parts[2]);
+                int income = Integer.parseInt(parts[3]);
+                if (parts[1].equals(name) && dateComparator(parts[0], dateFrom, dateTo)) {
+                    salary += hours * income;
+                }
+            }
+            result.append(System.lineSeparator()).append(name).append(" - ").append(salary);
+        }
+
+        return String.valueOf(result);
```

Comment:
There's no need to call String.valueOf on a StringBuilder instance. StringBuilder's toString method provides the necessary functionality to convert it to a String.